### ISSUE TITLE
Windows xibo client upgrade from 1.6.4 > 1.7.x

### ISSUE THREAD

**Issue by scienceworld** [xibosignage/xibo#372]
Upgrading windows clients from 1.6.4 to 1.7.x

1) when the CMS is upgraded 1.6.4 clients stop showing their content
2) when you push the MSI to workstations running 1.6.4 the client settings aren't retained, including server url, key and path.  When upgrading a large qty of displays via windows group policy, it would be nice for the client to retain the settings
3) in addition to the client retaining settings from a v1.6.4 previous install, it would also be nice to be able to supply a cfg file or creating an MST transform with the key, url and path parameters specified so the client used those when launching the first time.

I guess this could be 3 issues, but they're all related to transitioning from 1.6.4 to 1.7

**Comment by scienceworld** [comment]
in addition to retaining the server settings and key, it would also be nice to retain the display ID.

this is particularly importing during an upgrade as all layouts and schedules would be assigned to the display id of the 1.6.4 installation.

**Comment by dasgarner** [comment]
Settings aren't managed player side any more, so you only need to create a display profile with the settings you require.

The hardware key is a tricky one - if you haven't changed it from installation it will be OK and continue to use the old one - if you have overridden it before it will create a new display (not much we can do about that unfortunately).

You could write something to look in the user profile at the `user.config` in the Xibo folder and update the config.xml file for the new version with the same hardware key.

**Comment by scienceworld** [comment]
Thanks for the feedback.

I was only looking for it to retain 4 settings from previous install
1) server key
2) server url
3) store path
4) hardware key

hardware key regeneration is problematic.  You can't change the hardware key for a display in the CMS, so if they player  generates a new key, you can't take over a display in the CMS.  The only other recourse is to use the old player hardware key on a new/upgraded install of the player on a machine.  due to natural lifecycles of displays its useful to be able to take over a previously comissioned display by impersonating the hardware key.

Is it not possible for the msi installer to load in the existing xibo config, parse those 4 values, and write the new config with them automatically?

**Comment by dasgarner** [comment]
> due to natural lifecycles of displays its useful to be able to take over a previously commissioned display by impersonating the hardware key.

You should still be able to do that, right? By setting the hardware key on the new player?


I totally agree with you that this is a pain - unfortunately we came under massive pressure to support multiple instances of the player running at once AND screen saver mode and this required a change to the way the config information is stored.

We won't be changing the way config information is stored in the future (unless we move to an entirely different player application - which will happen eventually).

I do not know if it is possible for the MSI installer to do it - I confess to using a wizard to package up the application.

It would be possible to make the Player itself do it - but it makes it unnecessarily complicated going forwards.

On balance, I think it might have to do a manual process for this one-time inconvenience.

**Comment by scienceworld** [comment]
I did manually upgrade the clients from 1.6.4 to 1.7.2 by manually replacing the hw key and manually populating the server url, server key and local folder.

It took some time to do this on 25 players.

I had to do the upgrade during a blackout period.  When the CMS was upgraded from 1.6.4 to 1.7.2 all existing 1.6.4 clients stopped playing their content.  We weren't able to do a rolling upgrade.

So, to gain traction with many clients I would recommend getting the MSI updater to take in the old config values and perform an upgrade vs a replacement install if possible.

J
